x/distribution/types/store.go

```diff
@@ -1,7 +1,26 @@
 package types
 
-const ()
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
 
-// store: per-validator deccoins, for bonded set
-// store: per-validator accumulated commission
-// store: storage records from withdrawals (fees / power)
+// historical rewards for a validator
+type ValidatorHistoricalRewards struct {
+	AccumulatedFees sdk.DecCoins `json:"accumulated_fees"`
+	TotalPower      sdk.Dec      `json:"total_power"`
+}
+
+// current rewards for a validator
+type ValidatorCurrentRewards struct {
+	Rewards sdk.DecCoins `json:"rewards"`
+	Period  uint64       `json:"period"`
+}
+
+// accumulated commission for a validator
+type ValidatorAccumulatedCommission = sdk.DecCoins
+
+// starting period for a delegator's rewards
+type DelegatorStartingPeriod = uint64
```

Comment:
This may break the Staking Inflation abstraction, but it would be ideal for this to be in the delegation struct.

Reply:
Hmm why? Saves one store read, but the better abstraction seems worth that.

Reply:
Its ideal in the sense that it (at least to me) feels like a property of the delegation. All accesses to a delegation have to do with fee distribution / rewards.

Reply:
Thats true. Another example where it would be bad is if someone used staking with a different fee distr. algorithm. In principle the SDK can handle this while maintaining the module abstractions. It _could_ have multiple modules share a particular subtree, such that both modules have an entry in every leaf location. I guess thats another #postlaunch item. (This is essentially the concept of column hashing)

Reply:
This actually will have high impact for the validator struct, since thats new store reads every block that can otherwise be a single store read. (since the leaf would be kv store cached) Per discussion with @sunnya97 this can also let us remove coins from being in the account, with the same efficiency, further helping improve abstraction barriers and efficient generalization. This seems to me like its really important to do. Probably should be converted into an issue